src/main/java/org/opensearch/neuralsearch/query/NeuralQueryBuilder.java

```diff
@@ -58,7 +58,7 @@
 @Accessors(chain = true, fluent = true)
 @NoArgsConstructor
 @AllArgsConstructor
-public class NeuralQueryBuilder extends AbstractQueryBuilder<NeuralQueryBuilder> {
+public class NeuralQueryBuilder extends AbstractQueryBuilder<NeuralQueryBuilder> implements ModelInferenceQueryBuilder {
```

Comment:
Here we use lombok.Setter and lombok.Getter for NeuralQueryBuilder and NeuralSparseQueryBuilder. It will generate the set&get method automatically for all fields.